### ISSUE TITLE
Investigate alternative drag and drop implementations

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#3478]
## What will this achieve?

Replaces `react-dnd` and restores drag and drop functionality in customize columns Datagrid extension.

## How will success be measured?

- [x] Both ES and CJS builds succeed
- [x] User can reorder columns via `CustomizeColumnsTearsheet`
- [x] Drag and drop follows the styles from the published [design guidance](https://pages.github.ibm.com/cdai-design/pal/patterns/drag-and-drop/usage)
- [x] User can reorder columns through keyboard actions
- [x] Tests are updated and pass
- [x] Determine whether to pursue a library or create our own implementation

## Additional information

- Designs
- Existing code
- etc

#3446

**Comment by elycheea** [comment]
@matthewgallo has been looking into [`react-beautiful-dnd`](https://github.com/atlassian/react-beautiful-dnd). So far seems like a viable option minus the drop zone styling. May need to discuss this with @carrenelloyd or Tomer if we want to pursue this one.

**Comment by lee-chase** [comment]
There are a number of options here, including creating our own. Looking at downloads it would appear that the big players are....

- react-draggable (keyboard - not in scope)
- react-beautiful-dnd (has keyboard options, maintenance not a priority) https://github.com/atlassian/react-beautiful-dnd#%EF%B8%8F-maintenance--support
- react-dnd (keyboard?? maintenance erratic and waiting on updates to be published) 
- dnd-kit (has keyboard access, created because react-beautiful-dnd is no longer maintained).

![image](https://github.com/carbon-design-system/ibm-products/assets/15086604/522bd9bb-7b96-4706-85a1-1aee20b7266f)

**Comment by elycheea** [comment]
From scrum with @lee-chase and @paul-balchin-ibm — we will investigate `react-beautiful-dnd` or `dnd-kit` for the moment to unblock customize columns with the understanding that this could be replaced when we pursue #2042
